resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -100,12 +100,18 @@ private[spark] class Client(
     kubernetesClient: KubernetesClient,
     watcher: LoggingPodStatusWatcher) extends Logging {
 
-  def run(): Unit = {
+  private[spark] def createConfigs(): (KubernetesDriverSpec, Map[String, String], ConfigMap) = {
```

Comment:
I guess this refeactor is for test, we'd better only separate configMap part, and then,  still keep resolvedDriverSpec in run, like this

```suggestion
  private[spark] def createConfigMap(
    driverSpec: KubernetesDriverSpec): (Map[String, String], ConfigMap) = {
    val configMapName = KubernetesClientUtils.configMapNameDriver
    val confFilesMap = KubernetesClientUtils.buildSparkConfDirFilesMap(configMapName,
      conf.sparkConf, driverSpec.systemProperties)
    val configMap = KubernetesClientUtils
      .buildConfigMap(configMapName, confFilesMap, conf.namespace)
    (confFilesMap, configMap)
  }

  def run(): Unit = {
    val resolvedDriverSpec = builder.buildFromFeatures(conf, kubernetesClient)
    val configMapName = KubernetesClientUtils.configMapNameDriver
    val (confFilesMap, configMap) = createConfigMap(resolvedDriverSpec)
```